prow/cmd/deck/static/script.js

```diff
@@ -61,12 +61,229 @@ function redrawOptions(opts) {
+}
+
 window.onload = function() {
+    document.addEventListener("keydown", function(event) {
+        if (event.keyCode === 40) {
+            handleDownKey();
+        } else if (event.keyCode === 38) {
+            handleUpKey();
+        }
+    });
     // set dropdown based on options from query string
-    redrawOptions(optionsForRepo(""));
+	var opts = optionsForRepo("");
+    initFuzzySearch(
+        "job",
+        "job-input",
+        "job-list",
+        Object.keys(opts["jobs"]).sort());
+    redrawOptions(opts);
     redraw();
 };
 
+function displayFuzzySearchResult(el, inputContainer) {
+    el.classList.add("active-fuzzy-search");
+    el.style.top = inputContainer.height + "px";
+    el.style.width = inputContainer.width + "px";
+    el.style.height = 200 + "px";
+}
+
+function fuzzySearch(id, inputId, listId, data, inputValue) {
+    var result = [];
+    inputValue = inputValue.replace("/^\s+|\s+$/g", "");
+
+    if (inputValue === "") {
+        addOptionFuzzySearch(data, id, inputId, listId, true);
+        return;
+    }
+
+    var reg = new RegExp(".*" + inputValue + ".*");
+    for (var i = 0; i < data.length; i++) {
+        if (reg.test(data[i])) {
+            result.push(data[i]);
+        }
+    }
+    addOptionFuzzySearch(result, id, inputId, listId, true);
+}
+
+function validToken(token) {
+    // 0-9
+    if (token >= 48 && token <= 57) return true;
+    // a-z
+    if (token >= 65 && token <= 90) return true;
+    // - and backspace
+    return token === 189 || token === 8;
+}
+
+function handleEnterKeyDown(inputId, listId) {
+    var input = document.getElementById(inputId);
+    var list = document.getElementById(listId);
+    if (!input || !list) return;
+
+    if (list.childElementCount === 0) return;
+
+    var selectedJobs = list.getElementsByClassName("job-selected");
+    var job = list.firstElementChild.innerHTML;
+    if (selectedJobs && selectedJobs.length === 1) {
+        job = selectedJobs[0].innerHTML;
+    }
+
+    input.value = job;
+    input.blur();
+    list.classList.remove("active-fuzzy-search");
+    redraw();
+}
+
+function registerFuzzySearchHandler(id, inputId, listId, data) {
+    var input = document.getElementById(inputId);
+    if (!input) return;
+
+    input.addEventListener("keydown", function() {
+		if (event.keyCode === 13) {
+			// If enter key is hit, seget the first job in the list.
+            handleEnterKeyDown(inputId, listId);
+		} else if (validToken(event.keyCode)) {
+		    // Delay 1 frame that the input character is recorded before getting
+            // input value
+            setTimeout(function () {
+                fuzzySearch(id, inputId, listId, data, input.value);
+            }, 32);
+		}
+
+    });
+}
+
+function initFuzzySearch(id, inputId, listId, data) {
+    var el = document.getElementById(id);
+    var input = document.getElementById(inputId);
+    var list = document.getElementById(listId);
+
+    if (!input || !list || !el) return;
```

Comment:
In what cases does this happen?

Reply:
only when people miss-type/ miss the id.